### PR TITLE
Stabilize CI

### PR DESCRIPTION
**Pull request by SabaunT**
Stabilizing mac-os builds

### CODE REVIEW

**Review by ukint-vs — APPROVED** [review]

**Comment by SabaunT** [comment]
This is proof the decision works [212373d](https://github.com/gear-tech/gear/pull/2416/commits/212373de5fa4ab26609487b90657b29a6299777a), now reverting some changes.

File: .github/workflows/CI.yaml [hunk 1]
```diff
@@ -11,7 +11,6 @@ on:
 
 env:
   CARGO_INCREMENTAL: 0
-  CARGO_REGISTRIES_CRATES_IO_PROTOCOL: sparse
```
Comment (shamilsan) [inline .github/workflows/CI.yaml:14]: I'd update toolchains to the latest version instead of moving backward.

Comment (SabaunT) [reply .github/workflows/CI.yaml:14]: the mac-os job installs somehow stable version of 1.67.1. Seems it downloads under the hood the rustup installer of that version. Dunno why not later version. I'd suggest removing until really needed the sparse/the later stable version is downloaded by default.

Comment (shamilsan) [reply .github/workflows/CI.yaml:14]: `actions-rs/toolchain` updates toolchain to the latest version. But up to you.

Comment (SabaunT) [reply .github/workflows/CI.yaml:14]: discussed.

**Pull request merged**